app/talk/lib/sidebar-notifications.cjsx

```diff
@@ -0,0 +1,53 @@
+React = require 'react'
+{Link} = require '@edpaget/react-router'
+talkClient = require '../../api/talk'
+resourceCount = require './resource-count'
+formatNumber = require './format-number'
+
+module?.exports = React.createClass
+  displayName: 'SidebarNotifications'
+
+  componentDidMount: ->
+    @getUndeliveredCount() if @props.user
+
+  componentWillReceiveProps: ->
+    @getUndeliveredCount() if @props.user
+
+  getInitialState: ->
+    { }
+
+  getUndeliveredCount: ->
+    delivered = false
+    page_size = 1
+    section = @props.section
+    section = "project-#{ @props.project.id }" if @props.project
+    talkClient.type('notifications').get({section, delivered, page_size}).then (notifications) =>
+      unreadCount = notifications[0]?.getMeta()?.count or 0
+      @setState {unreadCount}
+
+  label: ->
+    if @state.unreadCount > 0
+      "Notifications (#{ formatNumber @state.unreadCount } unread)"
+    else
+      'Notifications'
+
+  render: ->
+    return <span></span> unless @props.user and @state.unreadCount?
```

Comment:
It's also safe to return `null` in these cases if you don't need a specific dom element rendered